rosdep/base.yaml

```diff
@@ -7160,6 +7160,19 @@ osmium:
   fedora: [libosmium-devel]
   nixos: [libosmium]
   ubuntu: [libosmium-dev]
+paho-mqtt-c:
```

Comment:
Typically we prefer that these match the names of the Ubuntu/Debian packages.  In this case, that would technically be `libpaho-mqtt1.3`, but I don't think we should encode the version into the package name.  Thus, I would suggest that we make this `libpaho-mqtt`, and the C++ one `libpaho-mqttpp`.

Reply:
I chose the same names that are already used for the releases for Noetic and Galactic, [see here](https://github.com/ros/rosdistro/blob/master/noetic/distribution.yaml#L6759). Do you still suggest the renaming?

Reply:
I have also added the development packages `libpaho-mqtt-dev` and `libpaho-mqttpp-dev` and thus changed the names of the library packages as you suggested.

Reply:
> In this case, that would technically be libpaho-mqtt1.3, but I don't think we should encode the version into the package name. Thus, I would suggest that we make this libpaho-mqtt, and the C++ one libpaho-mqttpp.

I would be interested in figuring out if this would be an acceptable general policy. It's cumbersome to change the version numbers in key names and it's slightly revolting that the key name may include a version number which differs from the version in another package, but what would be done in the case of a debian distribution where there are multiple versions of the library package? Which version would get the unversioned rosdep key and how would we decide that?

I lean heavily towards verbatim debian package names, my internal hesitation is motivated by the fact that the version number is likely incorrect for other platforms much more than it being annoying to update or juggle keys. @tfoote would you be willing to weigh in?

Reply:
to ensure the lowest amount of friction for those upgrading from Noetic and Galactic, indeed do we not want to just preserve the stability of these existing names?

Reply:
> to ensure the lowest amount of friction for those upgrading from Noetic and Galactic, indeed do we not want to just preserve the stability of these existing names?

Personally, I am also leaning towards this solution. This would also allow us to easily support ROS2 Galactic and ROS2 Humble+ at the same time with our [mqtt_client[(https://github.com/ika-rwth-aachen/mqtt_clienthttps://github.com/ika-rwth-aachen/mqtt_client), which is currently blocked by this PR.

Reply:
> to ensure the lowest amount of friction for those upgrading from Noetic and Galactic, indeed do we not want to just preserve the stability of these existing names?

I don't think this is something we have to worry about here.  In Noetic, these were released as "vendor" style packages, meaning that they would be something like `ros-noetic-paho-mqtt-c`, with a rosdep key of `paho-mqtt-c`.  These a rosdep keys, which use a different convention (which we are debating here).  I think it is actually a good thing that we are changing that name; it will be clearer that it isn't exactly the same thing as what came from ROS 1.